packages/services/src/terminal/default.ts

```diff
@@ -228,9 +237,35 @@ class DefaultTerminalSession implements TerminalSession.ISession {
           reject(event);
         }
       };
+
+      socket.onclose = (event: CloseEvent) => {
+        console.error(`Terminal websocket closed: ${event.code}`);
+        this._reconnectSocket();
+      };
     });
   }
 
+  private _reconnectSocket = (): void => {
+    if (this._isDisposed || !this._ws) {
+      return;
+    }
+    if (this._autoReconnectAttempt < this._autoReconnectLimit) {
+      this._isReady = false;
+      let timeout = Math.pow(2, this._autoReconnectAttempt);
```

Comment:
Should this have an upper bound so that the it keeps attempting to reconnect at the very last at that maximum timeout?